### PR TITLE
[FLOC-1629] Release 0.4.0

### PR DESCRIPTION
**Pull request by adamtheturtle**
Fixes FLOC-1629.
This branch should not be merged until the release process is complete.

### CODE REVIEW

File: NEWS [hunk 1]
```diff
@@ -1,5 +1,5 @@
-Flocker 0.4.0pre1 (2015-03-24)
-==============================
+Flocker 0.4.0 (2015-04-07)
+==========================
 
 - New getting started guide. (FLOC-1261)
```
Comment (tomprince) [inline NEWS]: This was already include in `0.3.2+doc1`, so I'm not sure if it worth mentioning here.

File: NEWS [hunk 2]
```diff
@@ -1,5 +1,5 @@
...
 - Updated to Twisted 15.0. (FLOC-1302)
```
Comment (tomprince) [inline NEWS]: I wonder if this is relevant to anybody, since we don't expose the fact that we are written in python.

File: NEWS [hunk 3]
```diff
@@ -15,6 +15,7 @@ Flocker 0.4.0pre1 (2015-03-24)
 - Flocker development machine now uses CentOS 7. (FLOC-1497)
 - Added instructions for installing ``flocker-node`` on CentOS 7. (FLOC-999)
 - Flocker documentation is now hosted on Amazon S3. (FLOC-561)
```
Comment (tomprince) [inline NEWS]: Nobody cares, I don't think.

**Comment by tomprince** [comment]
There was some internal discussion about how the new API should be publicized. I think the current NEWS/whats-new suggests that it is in a more usable state than it actually is. Marketing should probably review the entire release-notes before the release proceedes.

I have completed the [pre-tag review](http://doc-dev.clusterhq.com/gettinginvolved/infrastructure/release-process.html#pre-tag-review-process), and other than the above comments about release-notes, the release process can proceed.

**Comment by adamtheturtle** [comment]
I'm not sure what NEWS should really include - the release process still says:

```
Ensure the release notes in NEWS are up-to-date:

XXX: Process to be decided. See https://clusterhq.atlassian.net/browse/FLOC-523
```

so I'll leave it for now, apart from that I don't want to wrongly publicise an API. I'll ask Marketing to check this.

**Comment by wallrj** [comment]
The release process documentation says
```
For a development release, the following redirects should work.

    https://docs.clusterhq.com/en/devel/ should redirect to https://docs.clusterhq.com/en/${VERSION}/
    https://docs.clusterhq.com/en/devel/authors.html should redirect to https://docs.clusterhq.com/en/${VERSION}/authors.html
```

But I think that instruction and the documentation publishing tool need updating to update the devel docs if the latest marketing release also happens to be the latest release.

Right now, the devel docs still point to the pre-release documentation, which isn't right.

**Comment by wallrj** [comment]
```
(release/flocker-0.4.0 =)[~/.../HybridLogic/flocker-0.4.0]$ curl -v https://docs.clusterhq.com/en/devel/authors.html >/dev/null
...
< Location: https://docs.clusterhq.com/en/0.4.0pre1/authors.html
...
```

**Comment by adamtheturtle** [comment]
> But I think that instruction and the documentation publishing tool need updating to update the devel docs if the latest marketing release also happens to be the latest release.

@wallrj I'm not convinced, nor do I have a strong opinion the other way. The routing rule starts as follows:

```
  <!-- Development Release Documentation -->
  <RoutingRule>
    <Condition>
      <KeyPrefixEquals>en/devel</KeyPrefixEquals>
```

and I don't see the value of /devel pointing to a marketing release.

Also, it is in my notes to make an issue to amen a tool which can check the redirects.

**Comment by wallrj** [comment]
The tag build results for my own reference:
 * http://build.clusterhq.com/boxes-flocker?branch=0.4.0

There are a some failures, but they're all known / expected intermittent issues:
 *  http://build.clusterhq.com/builders/flocker%2Finstalled-package%2Ffedora-20/builds/1488/steps/trial/logs/stdio

**Comment by wallrj** [comment]
Some new warnings when running the `linux-install.sh` script in a Fedora20 docker container:

```
docker run --tty --interactive --volume $PWD:/flocker fedora:20 /bin/bash
```

```
[root@1efeec9e5c6c ~]# sh linux-install.sh
Running virtualenv with interpreter /usr/bin/python2.7
New python executable in flocker-tutorial/bin/python2.7
Also creating executable in flocker-tutorial/bin/python
Installing setuptools, pip...done.
Downloading/unpacking pip from https://pypi.python.org/packages/py2.py3/p/pip/pip-6.1.1-py2.py3-none-any.whl#md5=172eb5abab25a5e0f7a7b63c7a49378d
  Downloading pip-6.1.1-py2.py3-none-any.whl (1.1MB): 1.1MB downloaded
Installing collected packages: pip
  Found existing installation: pip 1.5.6
    Uninstalling pip:
      Successfully uninstalled pip
Successfully installed pip
Cleaning up...
Installing Flocker and dependencies, this may take a few minutes with no output to the terminal...
/root/flocker-tutorial/lib/python2.7/site-packages/pip/_vendor/requests/packages/urllib3/util/ssl_.py:79: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. For more information, see https://urllib3.readthedocs.org/en/latest/security.html#insecureplatformwarning.
  InsecurePlatformWarning
/root/flocker-tutorial/lib/python2.7/site-packages/pip/_vendor/requests/packages/urllib3/util/ssl_.py:79: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. For more information, see https://urllib3.readthedocs.org/en/latest/security.html#insecureplatformwarning.
  InsecurePlatformWarning
/root/flocker-tutorial/lib/python2.7/site-packages/pip/_vendor/requests/packages/urllib3/util/ssl_.py:79: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. For more information, see https://urllib3.readthedocs.org/en/latest/security.html#insecureplatformwarning.
  InsecurePlatformWarning
Done!
[root@1efeec9e5c6c ~]# cd flocker-tutorial
[root@1efeec9e5c6c flocker-tutorial]# bin/flocker-deploy --version
0.4.0
````


These might be unique to the Docker Fedora image, but we should probably investigate and suppress them if we think they're not important or fix them or pin to an older requests package which doesn't issue those warnings, maybe.

**Comment by adamtheturtle** [comment]
> These might be unique to the Docker Fedora image

@wallrj I saw them on the boxcutter Vagrant image and had a note to report it.

**Comment by wallrj** [comment]
The encrypted vagrant tutorial box URL for the box on Atlas is redirecting to an insecure link on google Storage.

```
(1599)(change-maximum-size-FLOC-1599 ? =)[~/.../HybridLogic/flocker]$ wget --spider https://atlas.hashicorp.com/clusterhq/boxes/flocker-tutorial/versions/0.4.0/providers/virtualbox.box
Spider mode enabled. Check if remote file exists.
--2015-04-09 18:02:34--  https://atlas.hashicorp.com/clusterhq/boxes/flocker-tutorial/versions/0.4.0/providers/virtualbox.box
Resolving atlas.hashicorp.com (atlas.hashicorp.com)... 54.175.82.169, 107.23.224.212
Connecting to atlas.hashicorp.com (atlas.hashicorp.com)|54.175.82.169|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: http://storage.googleapis.com/clusterhq-vagrant/flocker-tutorial-0.4.0.box [following]
Spider mode enabled. Check if remote file exists.
--2015-04-09 18:02:35--  http://storage.googleapis.com/clusterhq-vagrant/flocker-tutorial-0.4.0.box
Resolving storage.googleapis.com (storage.googleapis.com)... 216.58.210.33, 2a00:1450:4009:800::2001
Connecting to storage.googleapis.com (storage.googleapis.com)|216.58.210.33|:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 1076132021 (1.0G) [application/vnd.previewsystems.box]
Remote file exists.
```

Release docs should note that the box url should be https....or we should automate the Atlas update steps.

**Comment by wallrj** [comment]
On trying to start the Vagrant tutorial I get some guest additions:

```
[/tmp/flocker-tutorial]$ vagrant up
Bringing machine 'node1' up with 'virtualbox' provider...
Bringing machine 'node2' up with 'virtualbox' provider...
==> node1: Box 'clusterhq/flocker-tutorial' could not be found. Attempting to find and install...
    node1: Box Provider: virtualbox
    node1: Box Version: = 0.4.0
==> node1: Loading metadata for box 'clusterhq/flocker-tutorial'
    node1: URL: https://atlas.hashicorp.com/clusterhq/flocker-tutorial
==> node1: Adding box 'clusterhq/flocker-tutorial' (v0.4.0) for provider: virtualbox
    node1: Downloading: https://atlas.hashicorp.com/clusterhq/boxes/flocker-tutorial/versions/0.4.0/providers/virtualbox.box
==> node1: Successfully added box 'clusterhq/flocker-tutorial' (v0.4.0) for 'virtualbox'!
==> node1: Importing base box 'clusterhq/flocker-tutorial'...
==> node1: Matching MAC address for NAT networking...
==> node1: Checking if box 'clusterhq/flocker-tutorial' is up to date...
==> node1: Setting the name of the VM: flocker-tutorial_node1_1428599358819_50613
==> node1: Clearing any previously set network interfaces...
==> node1: Preparing network interfaces based on configuration...
    node1: Adapter 1: nat
    node1: Adapter 2: hostonly
==> node1: Forwarding ports...
    node1: 22 => 2222 (adapter 1)
==> node1: Booting VM...
==> node1: Waiting for machine to boot. This may take a few minutes...
    node1: SSH address: 127.0.0.1:2222
    node1: SSH username: vagrant
    node1: SSH auth method: private key
    node1: Warning: Connection timeout. Retrying...
    node1: Warning: Remote connection disconnect. Retrying...
    node1: Warning: Remote connection disconnect. Retrying...
    node1: Warning: Remote connection disconnect. Retrying...
    node1: 
    node1: Vagrant insecure key detected. Vagrant will automatically replace
    node1: this with a newly generated keypair for better security.
    node1: 
    node1: Inserting generated public key within guest...
    node1: Removing insecure key from the guest if its present...
    node1: Key inserted! Disconnecting and reconnecting using new SSH key...
==> node1: Machine booted and ready!
GuestAdditions versions on your host (4.3.26) and guest (4.3.20) do not match.
The guest's platform is currently not supported, will try generic Linux method...
Copy iso file /usr/share/virtualbox/VBoxGuestAdditions.iso into the box /tmp/VBoxGuestAdditions.iso
mount: /dev/loop2 is write-protected, mounting read-only
Installing Virtualbox Guest Additions 4.3.26 - guest version is 4.3.20
Verifying archive integrity... All good.
Uncompressing VirtualBox 4.3.26 Guest Additions for Linux............
VirtualBox Guest Additions installer
Removing installed version 4.3.20 of VirtualBox Guest Additions...
Copying additional installer modules ...
Installing additional modules ...
Removing existing VirtualBox DKMS kernel modules[  OK  ]
Removing existing VirtualBox non-DKMS kernel modules[  OK  ]
Building the VirtualBox Guest Additions kernel modules[  OK  ]
Doing non-kernel setup of the Guest Additions[  OK  ]
You should restart your guest to make sure the new modules are actually used

Installing the Window System drivers
Could not find the X.Org or XFree86 Window System, skipping.
An error occurred during installation of VirtualBox Guest Additions 4.3.26. Some functionality may not work as intended.
In most cases it is OK that the "Window System drivers" installation failed.
==> node1: Checking for guest additions in VM...
==> node1: Setting hostname...
==> node1: Configuring and enabling network interfaces...
==> node1: Mounting shared folders...
    node1: /vagrant => /tmp/flocker-tutorial
==> node2: Box 'clusterhq/flocker-tutorial' could not be found. Attempting to find and install...
    node2: Box Provider: virtualbox
    node2: Box Version: = 0.4.0
==> node2: Loading metadata for box 'clusterhq/flocker-tutorial'
    node2: URL: https://atlas.hashicorp.com/clusterhq/flocker-tutorial
==> node2: Adding box 'clusterhq/flocker-tutorial' (v0.4.0) for provider: virtualbox
==> node2: Importing base box 'clusterhq/flocker-tutorial'...
==> node2: Matching MAC address for NAT networking...
==> node2: Checking if box 'clusterhq/flocker-tutorial' is up to date...
==> node2: Setting the name of the VM: flocker-tutorial_node2_1428599579485_97221
==> node2: Fixed port collision for 22 => 2222. Now on port 2200.
==> node2: Clearing any previously set network interfaces...
==> node2: Preparing network interfaces based on configuration...
    node2: Adapter 1: nat
    node2: Adapter 2: hostonly
==> node2: Forwarding ports...
    node2: 22 => 2200 (adapter 1)
==> node2: Booting VM...
==> node2: Waiting for machine to boot. This may take a few minutes...
    node2: SSH address: 127.0.0.1:2200
    node2: SSH username: vagrant
    node2: SSH auth method: private key
    node2: Warning: Connection timeout. Retrying...
    node2: Warning: Remote connection disconnect. Retrying...
    node2: Warning: Remote connection disconnect. Retrying...
    node2: Warning: Remote connection disconnect. Retrying...
    node2: 
    node2: Vagrant insecure key detected. Vagrant will automatically replace
    node2: this with a newly generated keypair for better security.
    node2: 
    node2: Inserting generated public key within guest...
    node2: Removing insecure key from the guest if its present...
    node2: Key inserted! Disconnecting and reconnecting using new SSH key...
==> node2: Machine booted and ready!
GuestAdditions versions on your host (4.3.26) and guest (4.3.20) do not match.
The guest's platform is currently not supported, will try generic Linux method...
Copy iso file /usr/share/virtualbox/VBoxGuestAdditions.iso into the box /tmp/VBoxGuestAdditions.iso
mount: /dev/loop2 is write-protected, mounting read-only
Installing Virtualbox Guest Additions 4.3.26 - guest version is 4.3.20
Verifying archive integrity... All good.
Uncompressing VirtualBox 4.3.26 Guest Additions for Linux............
VirtualBox Guest Additions installer
Removing installed version 4.3.20 of VirtualBox Guest Additions...
Copying additional installer modules ...
Installing additional modules ...
Removing existing VirtualBox DKMS kernel modules[  OK  ]
Removing existing VirtualBox non-DKMS kernel modules[  OK  ]
Building the VirtualBox Guest Additions kernel modules[  OK  ]
Doing non-kernel setup of the Guest Additions[  OK  ]
You should restart your guest to make sure the new modules are actually used

Installing the Window System drivers
Could not find the X.Org or XFree86 Window System, skipping.
An error occurred during installation of VirtualBox Guest Additions 4.3.26. Some functionality may not work as intended.
In most cases it is OK that the "Window System drivers" installation failed.
==> node2: Checking for guest additions in VM...
==> node2: Setting hostname...
==> node2: Configuring and enabling network interfaces...
==> node2: Mounting shared folders...
    node2: /vagrant => /tmp/flocker-tutorial

```

**Comment by wallrj** [comment]
I don't think the 0.4.0 RPM files have made it onto our S3 repo:

```
[/tmp/flocker-tutorial]$ cat /etc/yum.repos.d/clusterhq.repo 
[clusterhq]
name=ClusterHQ for Fedora $releasever
baseurl=https://s3.amazonaws.com/clusterhq-archive/fedora/$releasever/$basearch/
enabled=1
gpgcheck=0

[clusterhq-testing]
name=ClusterHQ development for Fedora $releasever
baseurl=https://s3.amazonaws.com/clusterhq-archive/fedora-testing/$releasever/$basearch/
enabled=0
gpgcheck=0

[/tmp/flocker-tutorial]$ repoquery --disablerepo='*' --enablerepo=clusterhq clusterhq-flocker-node
clusterhq-flocker-node-0:0.3.3-0.dev.8.627.gfb8bb23.noarch

[/tmp/flocker-tutorial]$ repoquery --disablerepo='*' --enablerepo=clusterhq-testing clusterhq-flocker-node
clusterhq-flocker-node-0:0.4.0-0.pre.1.noarch
```

**Comment by wallrj** [comment]
Yeah, there's no sign of the 0.4.0 package

```
[/tmp/flocker-tutorial]$ repoquery --show-duplicates --disablerepo='*' --enablerepo=clusterhq-testing '*'
clusterhq-flocker-cli-0:0.3.3-0.dev.7.noarch
clusterhq-flocker-cli-0:0.4.0-0.pre.1.noarch
clusterhq-flocker-node-0:0.3.3-0.dev.7.noarch
clusterhq-flocker-node-0:0.4.0-0.pre.1.noarch
clusterhq-python-flocker-0:0.3.3-0.dev.7.x86_64
clusterhq-python-flocker-0:0.4.0-0.pre.1.x86_64
```

```
[/tmp/flocker-tutorial]$ repoquery --show-duplicates --disablerepo='*' --enablerepo=clusterhq '*'
libnvpair1-0:0.6.3-235_g7f3e466.fc20.x86_64
libuutil1-0:0.6.3-235_g7f3e466.fc20.x86_64
libzfs2-0:0.6.3-235_g7f3e466.fc20.x86_64
libzfs2-devel-0:0.6.3-235_g7f3e466.fc20.x86_64
libzpool2-0:0.6.3-235_g7f3e466.fc20.x86_64
spl-0:0.6.3-76_g6ab0866.fc20.x86_64
spl-debuginfo-0:0.6.3-76_g6ab0866.fc20.x86_64
spl-dkms-0:0.6.3-76_g6ab0866.fc20.noarch
zfs-0:0.6.3-235_g7f3e466.fc20.x86_64
zfs-debuginfo-0:0.6.3-235_g7f3e466.fc20.x86_64
zfs-devel-0:0.6.2-287_g2024041.fc20.x86_64
zfs-dkms-0:0.6.3-235_g7f3e466.fc20.noarch
zfs-dracut-0:0.6.3-235_g7f3e466.fc20.x86_64
zfs-release-0:1-3.fc20.noarch
zfs-test-0:0.6.3-235_g7f3e466.fc20.x86_64

```

**Comment by adamtheturtle** [comment]
```
Installing the Window System drivers
Could not find the X.Org or XFree86 Window System, skipping.
An error occurred during installation of VirtualBox Guest Additions 4.3.26. Some functionality may not work as intended.
In most cases it is OK that the "Window System drivers" installation failed.
```

I've seen that a lot in multiple boxes and haven't encountered a problem that seems to be related.

**Comment by adamtheturtle** [comment]
> Yeah, there's no sign of the 0.4.0 package

I re-ran the script and it is fine. Unfortunately I don't have my history, but I was pretty sure I ran the script. Let's test for this next time and if it doesn't happen assume that I missed a command out.

**Comment by adamtheturtle** [comment]
I followed https://docs.clusterhq.com/en/0.4.0/indepth/installation.html?highlight=centos#installing-on-centos-7 on `boxcutter/centos70` with Vagrant and it worked - as far as I got `flocker-reportstate --version` to output 0.4.0.

**Comment by wallrj** [comment]
Ok. @adamtheturtle has re-run the package upload script and the packages are now visible in both the Fedora20 and Centos7 repos:

```
[/tmp/flocker-tutorial]$ repoquery --location --show-duplicates --repofrompath=clusterhq-centos7,https://s3.amazonaws.com/clusterhq-archive/centos/7/x86_64 --disablerepo='*' --enablerepo=clusterhq-centos7 '*'
https://s3.amazonaws.com/clusterhq-archive/centos/7/x86_64/clusterhq-flocker-cli-0.4.0-1.noarch.rpm
https://s3.amazonaws.com/clusterhq-archive/centos/7/x86_64/clusterhq-flocker-node-0.4.0-1.noarch.rpm
https://s3.amazonaws.com/clusterhq-archive/centos/7/x86_64/clusterhq-python-flocker-0.4.0-1.x86_64.rpm
https://s3.amazonaws.com/clusterhq-archive/centos/7/x86_64/docker-1.5.0-28.el7.centos.chq.2.x86_64.rpm
```

```
[/tmp/flocker-tutorial]$ repoquery --location --show-duplicates --repofrompath=clusterhq-fedora20,https://s3.amazonaws.com/clusterhq-archive/fedora/20/x86_64 --disablerepo='*' --enablerepo=clusterhq-fedora20 '*'
https://s3.amazonaws.com/clusterhq-archive/fedora/20/x86_64/clusterhq-flocker-cli-0.4.0-1.noarch.rpm
https://s3.amazonaws.com/clusterhq-archive/fedora/20/x86_64/clusterhq-flocker-node-0.4.0-1.noarch.rpm
https://s3.amazonaws.com/clusterhq-archive/fedora/20/x86_64/clusterhq-python-flocker-0.4.0-1.x86_64.rpm
https://s3.amazonaws.com/clusterhq-archive/fedora/20/x86_64/libnvpair1-0.6.3-235_g7f3e466.fc20.x86_64.rpm
https://s3.amazonaws.com/clusterhq-archive/fedora/20/x86_64/libuutil1-0.6.3-235_g7f3e466.fc20.x86_64.rpm
https://s3.amazonaws.com/clusterhq-archive/fedora/20/x86_64/libzfs2-0.6.3-235_g7f3e466.fc20.x86_64.rpm
https://s3.amazonaws.com/clusterhq-archive/fedora/20/x86_64/libzfs2-devel-0.6.3-235_g7f3e466.fc20.x86_64.rpm
https://s3.amazonaws.com/clusterhq-archive/fedora/20/x86_64/libzpool2-0.6.3-235_g7f3e466.fc20.x86_64.rpm
https://s3.amazonaws.com/clusterhq-archive/fedora/20/x86_64/spl-0.6.3-76_g6ab0866.fc20.x86_64.rpm
https://s3.amazonaws.com/clusterhq-archive/fedora/20/x86_64/spl-debuginfo-0.6.3-76_g6ab0866.fc20.x86_64.rpm
https://s3.amazonaws.com/clusterhq-archive/fedora/20/x86_64/spl-dkms-0.6.3-76_g6ab0866.fc20.noarch.rpm
https://s3.amazonaws.com/clusterhq-archive/fedora/20/x86_64/zfs-0.6.3-235_g7f3e466.fc20.x86_64.rpm
https://s3.amazonaws.com/clusterhq-archive/fedora/20/x86_64/zfs-debuginfo-0.6.3-235_g7f3e466.fc20.x86_64.rpm
https://s3.amazonaws.com/clusterhq-archive/fedora/20/x86_64/zfs-devel-0.6.2-287_g2024041.fc20.x86_64.rpm
https://s3.amazonaws.com/clusterhq-archive/fedora/20/x86_64/zfs-dkms-0.6.3-235_g7f3e466.fc20.noarch.rpm
https://s3.amazonaws.com/clusterhq-archive/fedora/20/x86_64/zfs-dracut-0.6.3-235_g7f3e466.fc20.x86_64.rpm
https://s3.amazonaws.com/clusterhq-archive/fedora/20/x86_64/zfs-release-1-3.fc20.noarch.rpm
https://s3.amazonaws.com/clusterhq-archive/fedora/20/x86_64/zfs-test-0.6.3-235_g7f3e466.fc20.x86_64.rpm

```

**Comment by wallrj** [comment]
@tomprince or @adamtheturtle Should the Centos7 repo also have copies of our custom ZFS on Linux packages? They don't appear to be there.

**Comment by wallrj** [comment]
Thanks @adamtheturtle.

Here's a summary of my testing, some issues with the documentation and some problems I encountered.

None of them are blockers so I'll merge.

Build results: http://build.clusterhq.com/boxes-flocker?branch=0.4.0
```
1. Documentation
   1. Redirects work
   2. TODO: Suggest updating devel doc links when marketing release is newer than last dev or pre release.

2. flocker-deploy installation
   1. Fedora20
      1. Installed successfully
      2. TODO: Suppress SSL warnings from requests library when running linux-install.sh
         https://github.com/ClusterHQ/flocker/pull/1297#issuecomment-91286249

   2. Ubuntu 14.04
      1. Installed successfully
         1. `docker run --tty --interactive --rm --hostname ubuntu1404 --volume $PWD:/flocker ubuntu:14.04 /bin/bash` (for future reference)
      2. TODO: Same SSL warnings as for Fedora20 (above)

   3. MacOSX
      1. Manually installed on Mac Mini
      2. Automated test also worked: http://build.clusterhq.com/builders/flocker%2Fhomebrew%2Ftest/builds/298/steps/run-homebrew-test/logs/stdio
      3. TODO: Ask about automatically triggering a test of the official homebrew recipe

3. flocker-node installation
   1. Repoquery
      1. Confirmed that the Fedora20 and Centos7 packages are now in our S3 repo
         1. `repoquery --show-duplicates --repofrompath=clusterhq-fedora20,https://s3.amazonaws.com/clusterhq-archive/fedora/20/x86_64 --disablerepo='*' --enablerepo=clusterhq-fedora20 '*'`
         2. `repoquery --show-duplicates --repofrompath=clusterhq-centos7,https://s3.amazonaws.com/clusterhq-archive/centos/7/x86_64 --disablerepo='*' --enablerepo=clusterhq-centos7 '*'`
      2. TODO: File an issue to suggest a Repoquery release or release-review step which ensures that the tagged package is available from our official repo.

   2. Fedora20
      1. Vagrant: https://docs.clusterhq.com/en/0.4.0/indepth/tutorial/vagrant-setup.html
         1. Tutorial environment boots up
         2. TODO: I had to run `vagrant plugin uninstall vagrant-vbguest` to get it working
            See https://github.com/ClusterHQ/flocker/pull/1297#issuecomment-91297191
         3. TODO: "You now have two VMs running and easy SSH access to them. This completes the Vagrant-related setup." but it doesn't link to the next step in the tutorial.
         4. TODO: The release review process doesn't test that packages can be installed from our official repo. Include new instruction to follow the Fedora20 installation instructions or create a builder that installs the Flocker packages from the clusterhq repo (rather than the buildbot repo).

      2. DigitalOcean: http://build.clusterhq.com/builders/flocker%2Facceptance%2Fdigitalocean%2Ffedora-20/builds/794
         1. Success!

      3. AWS:
         1. Manually installed 2 x flocker-node on Fedora20 on AWS. Then completed the ELK tutorial.
         2. TODO: File issue to add a F20 AWS builder
         3. TODO: There should be a link beneath the Fedora20 and Centos7 specific instructions saying "now go to the generic Fedora20 / Centos7 post installation instructions"
         4. TODO: Bad wording "You have now installed clusterhq-flocker-node and created a ZFS for it."
         5. TODO: Bad wording "Keep in mind that the tutorial was designed with local virtual machines in mind, and results in an insecure environment."

      4. Rackspace: http://build.clusterhq.com/builders/flocker%2Facceptance%2Frackspace%2Ffedora-20/builds/695
         1. Success.

   3. Centos7
      1. Vagrant:
         1. Successfully installed on a chef/centos-7.0 vagrant box.
         2. TODO: The release process doesn't include an instruction to follow the Centos7 installation instructions. Either add those instructions or create a builder that installs the Flocker packages from the clusterhq repo (rather than the buildbot repo).
         3. TODO: Propose that Centos7 vagrant-tutorial-box is high priority and add comment to issue.
         4. TODO: There should be an acceptance test runner for a vanilla installation of Centos7 (if there is such a thing). All the cloud images seem to have non-standard default settings, so it would be good to see that our post-installation Centos7 instructions would work on a standard installation.
         5. TODO: Centos7 installation instructions fail if ZFS is already installed and the kernel is upgraded. ZFS modules were not recompiled automatically.

      2. AWS: http://build.clusterhq.com/builders/flocker%2Facceptance%2Faws%2Fcentos-7/builds/751
         1. Success

      3. Rackspace: http://build.clusterhq.com/builders/flocker%2Facceptance%2Frackspace%2Fcentos-7/builds/742
         1. Success

      4. DigitalOcean:
         1. TODO: File an issue for automatic Centos7 acceptance tests on Digital Ocean
```

**Comment by wallrj** [comment]
PS I updated the Atlas link to be https and the box still seems to download.

```
[/tmp/flocker-tutorial]$ wget --spider https://atlas.hashicorp.com/clusterhq/boxes/flocker-tutorial/versions/0.4.0/providers/virtualbox.box
Spider mode enabled. Check if remote file exists.
--2015-04-10 15:49:52--  https://atlas.hashicorp.com/clusterhq/boxes/flocker-tutorial/versions/0.4.0/providers/virtualbox.box
Resolving atlas.hashicorp.com (atlas.hashicorp.com)... 54.175.82.169, 107.23.224.212
Connecting to atlas.hashicorp.com (atlas.hashicorp.com)|54.175.82.169|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://storage.googleapis.com/clusterhq-vagrant/flocker-tutorial-0.4.0.box [following]
Spider mode enabled. Check if remote file exists.
--2015-04-10 15:49:57--  https://storage.googleapis.com/clusterhq-vagrant/flocker-tutorial-0.4.0.box
Resolving storage.googleapis.com (storage.googleapis.com)... 216.58.210.33, 2a00:1450:4009:800::2001
Connecting to storage.googleapis.com (storage.googleapis.com)|216.58.210.33|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 1076132021 (1.0G) [application/vnd.previewsystems.box]
Remote file exists.
```

```
[/tmp/flocker-tutorial]$ vagrant up
Bringing machine 'node1' up with 'virtualbox' provider...
Bringing machine 'node2' up with 'virtualbox' provider...
==> node1: Box 'clusterhq/flocker-tutorial' could not be found. Attempting to find and install...
    node1: Box Provider: virtualbox
    node1: Box Version: = 0.4.0
==> node1: Loading metadata for box 'clusterhq/flocker-tutorial'
    node1: URL: https://atlas.hashicorp.com/clusterhq/flocker-tutorial
==> node1: Adding box 'clusterhq/flocker-tutorial' (v0.4.0) for provider: virtualbox
    node1: Downloading: https://atlas.hashicorp.com/clusterhq/boxes/flocker-tutorial/versions/0.4.0/providers/virtualbox.box

```

**Pull request merged**